goldmane/pkg/aggregator/ring_index.go

```diff
@@ -16,39 +16,46 @@ package aggregator
 
 import (
 	"fmt"
+	"math"
 	"sort"
 
 	"github.com/sirupsen/logrus"
 
 	"github.com/projectcalico/calico/goldmane/pkg/internal/types"
+	"github.com/projectcalico/calico/libcalico-go/lib/set"
 )
 
-func NewRingIndex(a *LogAggregator) *RingIndex {
+type logAggregator interface {
```

Comment:
I think it's worth adding a comment or two about why we have this intermediary interface.

Reply:
done